src/Modal.js

```diff
@@ -54,6 +54,7 @@ const propTypes = {
     PropTypes.string,
     PropTypes.func,
   ]),
+  mountNode: PropTypes.any,
```

Comment:
We should probably limit this like we do for the `target`s in tooltip/popover and allow for a DOMElement, function (which will return a DOMElement), or a string (which can be used to find a DOMElement).

There is a util which handles all of those cases, returning the DOMElement if found.

Probably should call it `container` to match similar functionality/props elsewhere.